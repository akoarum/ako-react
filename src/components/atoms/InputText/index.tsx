import styled from 'styled-components'
import * as React from 'react'
import * as style from '../../../utils/style'

type Props = {
  className?: string
  type: string
  value: string
  id?: string
  placeholder?: string
  maxLength?: number
  min?: number
  max?: number
  required?: boolean
  onInput?: (e: React.FormEvent<HTMLInputElement>) => void
  onFocus?: (e: React.FormEvent<HTMLInputElement>) => void
  onBlur?: (e: React.FormEvent<HTMLInputElement>) => void
}

export const StyledInputText = styled.input`
  box-sizing: border-box;
  border: 1px solid ${style.color.border};
  padding: 8px 12px;
  width: 100%;
  font-size: ${style.font.sizePc};
  outline: none;
  transition: border 0.3s ${style.animation.easing};

  &:focus {
    border: 1px solid ${style.color.theme};
  }

  &[aria-invalid='true'] {
    border: 1px solid ${style.color.caution};
  }
`

export const InputText: React.FC<Props> = props => {
  return (
    <StyledInputText
      id={props.id}
      type={props.type}
      defaultValue={props.value}
      placeholder={props.placeholder}
      maxLength={props.maxLength}
      min={props.min}
      max={props.max}
      className={props.className}
      required={props.required}
      onInput={props.onInput}
      onFocus={props.onFocus}
      onBlur={props.onBlur}
    />
  )
}

export default InputText
