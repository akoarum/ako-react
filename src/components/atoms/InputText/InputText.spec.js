import React from 'react'
import renderer from 'react-test-renderer'
import InputText from './index'

describe('InputText', () => {
  let wrapper

  beforeEach(() => {
    wrapper = renderer.create(
      <InputText />
    )
  })

  describe('initialize', () => {
    it('snapshot', () => {
      const tree = wrapper.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
