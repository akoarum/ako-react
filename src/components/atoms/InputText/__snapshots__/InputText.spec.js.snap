// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText initialize snapshot 1`] = `
<input
  className="sc-bdVaJa lbiXnV"
/>
`;
